src/librustc/ty/mod.rs

```diff
@@ -2380,6 +2380,21 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
     }
 }
 
+impl<'a, 'tcx> TyCtxt<'a, 'tcx, 'tcx> {
+    pub fn is_sized(self, ty: Ty<'tcx>) -> bool {
```

Comment:
Ideally these two methods would be avoided. I wonder what the `DUMMY_SP` is used for - `is_sized` should never cause an ICE or an user error.

Reply:
I wasn't sure about those functions, there are still a bunch of them in `rustc_trans::common.rs`. Initially I thought I would move them into `TyCtxt` but then I asked myself if they should even exist. Should I remove them?

Reply:
Depends on what the usage looks like.

Reply:
I haven't investigated it so far. These functions were just copied (not moved) into `trans_utils::common.rs` and therefore are only used in the collector IIRC. But the original functions are probably more widespread.

Reply:
Replacing the uses in the collector with the more common forms is probably the best first step.